--- dosemu-1.4.0.8.orig/etc/dosemu.conf
+++ dosemu-1.4.0.8/etc/dosemu.conf
@@ -72,6 +72,7 @@
 # be mixed such as "hdimage_c /dev/hda1 /dev/hda3:ro". Default: "drives/*"

 # $_hdimage = "drives/*"
+$_hdimage = "bootdir"

 # if you want to boot from a virtual floppy:
 # file name of the floppy image under DOSEMU_LIB_DIR
@@ -309,6 +310,7 @@
 # console "$_graphics=(1)" mode, but GPM can still be used in other modes).

 # $_mouse_dev = ""
+$_mouse_dev = "/dev/mouse"

 # Default: "" or one or more of: "emulate3buttons cleardtr"

--- dosemu-1.4.0.8.orig/etc/dosemu.users.example
+++ dosemu-1.4.0.8/etc/dosemu.users.example
@@ -14,7 +14,7 @@
 #
 # For more details look at README.txt

-#default_lib_dir= /usr/local/share/dosemu
+default_lib_dir= /var/lib/dosemu
 #config_script= builtin # (or e.g. /etc/dosemu/global.conf)
 #log_level= 0
 #carol is allowed to do do all weird things
