--- dosemu-1.1.4/src/base/init/parser.y.in.orig	Wed Dec 18 02:36:44 2002
+++ dosemu-1.1.4/src/base/init/parser.y.in	Sat Jan 25 22:52:12 2003
@@ -2576,7 +2576,7 @@
 	       pwd->pw_name);
        if(log_syslog>=1) {
 	   fprintf(stderr, "This event will be logged!\n");
-           sprintf(buf, "Illegal DOSEMU start attempt by %s (uid=%i)", 
+           snprintf(buf, sizeof(buf), "Illegal DOSEMU start attempt by %s (uid=%i)", 
            pwd->pw_name, uid);
 	   write_to_syslog(buf);
        }
@@ -2584,7 +2584,7 @@
   }
   else {
     if(log_syslog>=2) {
-       sprintf(buf, "DOSEMU started%s by %s (uid/euid=%i/%i)",
+       snprintf(buf, sizeof(buf), "DOSEMU started%s by %s (uid/euid=%i/%i)",
             (can_do_root_stuff && !under_root_login)? " suid root" : "",
             pwd->pw_name, uid, get_orig_euid());
        write_to_syslog(buf);
