--- dosemu-1.1.3/src/plugin/kbd_unicode/serv_xlat.c.orig	Thu Jun 20 16:20:54 2002
+++ dosemu-1.1.3/src/plugin/kbd_unicode/serv_xlat.c	Thu Jun 20 19:57:10 2002
@@ -1723,8 +1723,8 @@
 	 */
 	bios_key = is_accent ? ascii : make_bios_code_r(make, key, ascii, keysym, state);
 
-	if (bios_key == 0x23e0)  /* Cyrillic_er work around */
-		bios_key &= 0x00FF;
+	if (bios_key == 0x23e0 || bios_key == 0x18e0)  /* Cyrillic_er work around */
+		bios_key &= 0x00FF;  /* ^^^^^^^^^^^^^^^^^^ Oacute */
 #if 0
 	k_printf("translate_key: keysym=%04x bios_key=%04x\n",
 		keysym, bios_key);
