--- dosemu-1.1.4/etc/dosemu.conf.orig	Wed Dec 18 02:36:44 2002
+++ dosemu-1.1.4/etc/dosemu.conf	Sat Jan 25 23:35:28 2003
@@ -27,7 +27,7 @@
 # Please edit only between the brackets and quotes and keep the rest intact.
 #        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^     ^^^^^^^^^^^^^^^^^^^^
 #
-#$_hdimage = "freedos"  
+$_hdimage = "bootdir"  
 #			# List of hdimages or boot directories under
 #                       # syshdimagedir (/var/lib/dosemu by default) 
 #			# assigned in this order such as
@@ -103,7 +103,7 @@
 #
 #$_rawkeyboard = (0)	# bypass normal keyboard input on the Linux console, 
 #			# maybe dangerous
-#$_layout = "auto"	# one of: finnish(-latin1), de(-latin1), be, it, us
+$_layout = "us"		# one of: finnish(-latin1), de(-latin1), be, it, us
 #			# uk, dk(-latin1), keyb-no, no-latin1, dvorak, po
 #			# sg(-latin1), fr(-latin1), sf(-latin1), es(-latin1)
 #			# sw, hu(-latin2), hu-cwi, keyb-user
@@ -159,7 +159,7 @@
 #
 #$_mouse = ""		# one of: microsoft, mousesystems, logitech, mmseries
 #			# mouseman, hitachi, busmouse, ps2, imps2
-#$_mouse_dev = ""	# one of: com1, com2, com3, com4 or /dev/mouse
+$_mouse_dev = "/dev/mouse"	# one of: com1, com2, com3, com4 or /dev/mouse
 #$_mouse_flags = ""	# empty list or one or more of:
 #			# "emulate3buttons cleardtr"
 #$_mouse_baud = (0)	# baudrate, 0 == don't set
