--- dosemu-1.4.0.8.orig/src/doc/DANG/Makefile
+++ dosemu-1.4.0.8/src/doc/DANG/Makefile
@@ -13,6 +13,8 @@

 all:    $(TXT) $(HTML)

+html:	$(HTML)
+
 $(TXT): %.txt: %.sgml
 	$(srcdir)/../tools/doSgmlTools.pl -t -v $<

--- dosemu-1.4.0.8.orig/src/doc/HOWTO/Makefile
+++ dosemu-1.4.0.8/src/doc/HOWTO/Makefile
@@ -13,6 +13,8 @@

 all:    $(TXT) $(HTML)

+html:	$(HTML)
+
 $(TXT): %.txt: %.sgml
 	$(srcdir)/../tools/doSgmlTools.pl -t $<

