diff -ur dosemu-1.0.2.orig/base-configure.in dosemu-1.0.2/base-configure.in
--- dosemu-1.0.2.orig/base-configure.in	Sun Jun 18 16:34:26 2000
+++ dosemu-1.0.2/base-configure.in	Thu Aug 17 09:56:30 2000
@@ -3855,19 +3855,8 @@
   OPT="$OPT -fomit-frame-pointer"  # never omit the frame pointer when debugging!
 fi
 
-# -m486 is usually in the specs for the compiler
-case "$host_cpu" in
-  i[[456]]86)
-    OPT="$OPT -m486 -malign-loops=2 -malign-jumps=2 -malign-functions=2"
-    ;;
-  i386)
-    OPT="$OPT -m386"
-    ;;
-  *)
-    ;;
-esac
 
-OPT="$OPT -MD" # this should give painless dependencies
+OPT="$OPTFLAGS -MD" # this should give painless dependencies
 
 # OPT=-fno-inline
 if test "$CONFIG_HOST" = "linux"; then
diff -ur dosemu-1.0.1.orig/man/Makefile dosemu-1.0.1/man/Makefile
--- dosemu-1.0.1.orig/man/Makefile	Sun Mar  5 13:57:36 2000
+++ dosemu-1.0.1/man/Makefile	Thu Aug 17 09:57:21 2000
@@ -7,6 +7,6 @@
 
 install:
 	@install -d $(INSTROOT)/usr/man/man1; \
-	install *dos*.1 mkfatimage*.1 $(INSTROOT)/usr/man/man1
+	install *dos*.1 mkfatimage*.1 $(INSTROOT)/usr/share/man/man1
 
 clean realclean:
diff -ur dosemu-1.0.2.orig/src/arch/linux/Makefile.main dosemu-1.0.2/src/arch/linux/Makefile.main
--- dosemu-1.0.2.orig/src/arch/linux/Makefile.main	Tue Mar 28 01:22:01 2000
+++ dosemu-1.0.2/src/arch/linux/Makefile.main	Thu Aug 17 10:02:53 2000
@@ -324,9 +324,9 @@
 	  install -m 0744 ../etc/keymap/* $(INSTROOT)/var/lib/dosemu/keymap; \
 	fi
 	@if [ -f $(INSTROOT)/lib/libemu ]; then rm -f $(INSTROOT)/lib/libemu ; fi
-	@install -c -o root -m 0755 $(BINPATH)/bin/$(DOSBIN) $(INSTROOT)/usr/bin
+	@install -c $(BINPATH)/bin/$(DOSBIN) $(INSTROOT)/usr/bin
 ifdef  USE_MHPDBG
-	@install -c -o root -m 0755 $(BINPATH)/bin/dosdebug $(INSTROOT)/usr/bin
+	@install -c $(BINPATH)/bin/dosdebug $(INSTROOT)/usr/bin
 endif
 ifndef ELF
 	@install -m 0644 $(LIBDOSEMU) $(INSTROOT)/usr/lib/libdosemu-$(THISVERSION)
