--- dosemu-1.0.0/base-configure.than	Sat May 27 11:36:48 2000
+++ dosemu-1.0.0/base-configure	Sat May 27 11:37:12 2000
@@ -3924,14 +3924,14 @@
   GCC_VERSION_CODE="-DGCC_VERSION_CODE=$GCC_VERSION_CODE"
 
   echo -n "checking for glibc..."
-  major=`echo '#include <features.h>' |${CC-cc} -E -dM -|awk '/#define.+__GLIBC__/{print $3}'`
+  major=`echo '#include <features.h>' |${CC-cc} -E -dM -|awk '/#define __GLIBC__ /{print $3}'`
   if test "$major" = ""; then
     echo "no"
   else
     if test "$(( $major > 1 ))" = "1"; then
       GLIBCFLAGS="-D_XOPEN_SOURCE"
     fi
-    minor=`echo '#include <features.h>' |${CC-cc} -E -dM -|awk '/#define.+__GLIBC_MINOR__/{print $3}'`
+    minor=`echo '#include <features.h>' |${CC-cc} -E -dM -|awk '/#define __GLIBC_MINOR__/{print $3}'`
     if test "$minor" = ""; then
       minor=0;
     fi
