--- dosemu-1.4.0/src/tools/periph/dexeconfig.c.orig	2008-10-10 16:01:05.802663404 +0200
+++ dosemu-1.4.0/src/tools/periph/dexeconfig.c	2008-10-10 16:15:55.241534296 +0200
@@ -238,7 +238,7 @@
     exit(1);
   }
   close(fd);
-  fc = open(cfile, O_WRONLY | O_CREAT | O_TRUNC);
+  fc = open(cfile, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
   if (fc < 0) {
     perror("cannot open config file");
     exit(1);
