--- dosemu-0.66.7/src/dosext/net/net/ipx.c.glibc	Mon Nov  3 16:42:48 1997
+++ dosemu-0.66.7/src/dosext/net/net/ipx.c	Mon Nov  3 16:45:34 1997
@@ -16,8 +16,8 @@
 #include <sys/types.h>
 #include <sys/vm86.h>
 #include <linux/sockios.h>
-#include <asm/types.h>
-#include <linux/ipx.h>
+#include <sys/types.h>
+#include <netipx/ipx.h>
 #ifndef IPX_TYPE
   #define IPX_TYPE 1   /* glibc's headers seem missing that
                           (normally defined in <linux/socket.h>) */
