--- dosemu-1.1.4/configure.ac.orig	Fri Mar 28 10:32:29 2003
+++ dosemu-1.1.4/configure.ac	Fri Mar 28 10:35:43 2003
@@ -421,15 +421,15 @@
 AC_SUBST(INCDIR)
 
 CFLAGS="-Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -Wpointer-arith"
+OPT="$OPTFLAGS"
 if test "$dodebug" = "yes"; then
   CFLAGS="$CFLAGS -gstabs+"
   AC_DEFINE(_DEBUG)
-  OPT="-O"
 else
-  OPT="-O2"   # BIG NOTE: do _not_ use -O3,
+# OPT="-O2"   # BIG NOTE: do _not_ use -O3,
               # because in forces _inline_ at places where we don't want it !
               # If we want _inline_, well then we 'code' it.
-  OPT="$OPT -fomit-frame-pointer"  # never omit the frame pointer when debugging!
+:#OPT="$OPT -fomit-frame-pointer"  # never omit the frame pointer when debugging!
 fi
 
 dnl CPU emulator
@@ -515,33 +515,6 @@
   fi
 
 
-  # -m486 is usually in the specs for the compiler
-  # but gcc-3.0 wants -mcpu=i486 instead
-  # let the compiler options be dependent of the _target_
-  # where dosemu is compiled for, not the host
-  if test "$target_cpu" = "athlon" -a $GCC_VERSION_CODE -le 2095 ;  then
-    target_cpu="i686"
-  fi
-  if test "$GCC_VERSION_CODE" -ge 2095 ; then
-     OPT="$OPT -mcpu=$target_cpu"
-  else
-     case "$target_cpu" in
-       i486)
-         OPT="$OPT -m486"
-         ;;
-       i386)
-         OPT="$OPT -m386"
-         ;;
-       *)
-         ;;
-     esac
-  fi
-  if test "$target_cpu" = "athlon"; then
-     OPT="$OPT -malign-functions=4 -fexpensive-optimizations"
-  elif test "$target_cpu" = "i486"; then
-     OPT="$OPT -malign-loops=2 -malign-jumps=2 -malign-functions=2"
-  fi
-
   echo -n "checking for glibc..."
   set `echo -e '#include <features.h>\nXXAaZZ __GLIBC__ XXBbZZ __GLIBC_MINOR__'|gcc -E -|awk '/XXAaZZ/ {print $2 " " $4}'`
   major=$1; minor=$2
--- dosemu-1.0.1.orig/man/Makefile	Sun Mar  5 13:57:36 2000
+++ dosemu-1.0.1/man/Makefile	Thu Aug 17 09:57:21 2000
@@ -7,6 +7,6 @@
 
 install:
 	@install -d $(INSTROOT)/usr/man/man1; \
-	install *dos*.1 mkfatimage*.1 $(INSTROOT)/usr/man/man1
+	install *dos*.1 mkfatimage*.1 $(INSTROOT)/usr/share/man/man1
 
 clean realclean:
--- dosemu-1.1.4/src/arch/linux/dosext/sound/midid/Makefile.orig	Wed Dec 18 02:36:44 2002
+++ dosemu-1.1.4/src/arch/linux/dosext/sound/midid/Makefile	Sun Jan 26 00:14:10 2003
@@ -15,7 +15,7 @@
 # No servicable parts below
 
 CC=gcc
-CFLAGS=-Wall -pipe -O -g
+CFLAGS=-Wall -pipe $(OPTFLAGS)
 LDFLAGS += -Wall
 OBJECTS=midid.o events.o io.o stats.o emulation.o device.o \
 oss.o midout.o timid.o null.o
