--- dosemu-1.1.4/etc/dosemu.conf.orig	Fri Mar 28 10:41:25 2003
+++ dosemu-1.1.4/etc/dosemu.conf	Fri Mar 28 10:43:53 2003
@@ -28,6 +28,7 @@
 #        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^     ^^^^^^^^^^^^^^^^^^^^
 #
 #$_hdimage = "drives/*"  
+$_hdimage = "bootdir"
 #			# List of hdimages or boot directories under
 #                       # syshdimagedir (/var/lib/dosemu by default) 
 #			# assigned in this order such as
@@ -158,7 +159,7 @@
 #
 #$_mouse = ""		# one of: microsoft, mousesystems, logitech, mmseries
 #			# mouseman, hitachi, busmouse, ps2, imps2
-#$_mouse_dev = ""	# one of: com1, com2, com3, com4 or /dev/mouse
+$_mouse_dev = "/dev/mouse"	# one of: com1, com2, com3, com4 or /dev/mouse
 #$_mouse_flags = ""	# empty list or one or more of:
 #			# "emulate3buttons cleardtr"
 #$_mouse_baud = (0)	# baudrate, 0 == don't set
--- dosemu-1.1.4/etc/dosemu.users.example.orig	Fri Mar 28 10:57:08 2003
+++ dosemu-1.1.4/etc/dosemu.users.example	Fri Mar 28 11:31:43 2003
@@ -14,7 +14,7 @@
 #
 # For more details look at ./doc/README.txt
 
-default_lib_dir= /usr/local/share/dosemu
+default_lib_dir= /var/lib/dosemu
 #config_script= builtin # (or e.g. /etc/dosemu/global.conf)
 log_level= 0
 #carol is allowed to do do all weird things
