diff -urN dosemu-1.1.5.org/configure.ac dosemu-1.1.5/configure.ac
--- dosemu-1.1.5.org/configure.ac	2003-09-01 23:10:03.000000000 +0200
+++ dosemu-1.1.5/configure.ac	2003-09-01 23:10:53.000000000 +0200
@@ -428,9 +428,9 @@
   if test "$enable_debug" = "yes"; then
     CFLAGS="$CFLAGS -g"
     AC_DEFINE(_DEBUG)
-    OPT="-O"
+    OPT="$OPTFLAGS"
   else
-    OPT="-O2" # BIG NOTE: do _not_ use -O3,
+    OPT="$OPTFLAGS" # BIG NOTE: do _not_ use -O3,
               # because in forces _inline_ at places where we don't want it !
               # If we want _inline_, well then we 'code' it.
     OPT="$OPT -fomit-frame-pointer"  # never omit the frame pointer when debugging!
