--- dosemu-1.1.5/src/base/bios/int10.c.orig	2003-07-19 17:21:55.000000000 +0200
+++ dosemu-1.1.5/src/base/bios/int10.c	2003-07-19 17:27:55.000000000 +0200
@@ -723,6 +723,7 @@
 /* helpers for font processing - eric@coli.uni-sb.de  11/2002 */
 /* only for TEXT mode: Otherwise, int 0x43 is used...         */
 
+#ifdef X_GRAPHICS
 static void vga_RAM_to_RAM(unsigned height, unsigned char chr, unsigned count,
                            unsigned seg, unsigned ofs, int bank)
 {
@@ -767,6 +768,7 @@
   }
   vga_RAM_to_RAM(height,0,256,seg,ofs,bank);
 }
+#endif
 
 /******************************************************************/
 
