--- dosemu-1.1.4/src/doc/DANG/Makefile.orig	Sat Feb 15 18:16:25 2003
+++ dosemu-1.1.4/src/doc/DANG/Makefile	Sat Feb 15 18:16:56 2003
@@ -9,6 +9,8 @@
 
 all:    $(TXT) $(HTML)
 
+html:	$(HTML)
+
 $(TXT): %.txt: %.sgml
 	../tools/doSgmlTools.pl -t -v $<
 
--- dosemu-1.1.4/src/doc/HOWTO/Makefile.orig	Sat Feb 15 18:17:14 2003
+++ dosemu-1.1.4/src/doc/HOWTO/Makefile	Sat Feb 15 18:17:34 2003
@@ -9,6 +9,8 @@
 
 all:    $(TXT) $(HTML)
 
+html:	$(HTML)
+
 $(TXT): %.txt: %.sgml
 	../tools/doSgmlTools.pl -t $<
 
