--- dosemu-1.0.1/src/dosext/mfs/mfs.c	Sun Mar  5 13:57:37 2000
+++ dosemu-1.0.1-p/src/dosext/mfs/mfs.c	Tue Jul 17 13:06:13 2001
@@ -171,6 +171,7 @@
 #include "bios.h"
 #endif
 
+#include <time.h>
 #include <stdio.h>
 #include <sys/file.h>
 #include <sys/ioctl.h>
--- dosemu-1.0.1/src/arch/linux/debugger/dosdebug.c	Sat May 20 20:58:12 2000
+++ dosemu-1.0.1-p/src/arch/linux/debugger/dosdebug.c	Tue Jul 17 13:16:01 2001
@@ -16,6 +16,7 @@
  * The switch-console code is from Kevin Buhr <buhr@stat.wisc.edu>
  */
 
+#include <time.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <sys/time.h>
