--- dosemu-1.0.1/src/base/keyboard/serv_xlat.c.orig	Mon Jul 24 00:07:15 2000
+++ dosemu-1.0.1/src/base/keyboard/serv_xlat.c	Mon Jul 24 00:08:15 2000
@@ -310,7 +310,7 @@
                   bios_scan=0x0f; ascii=0; break;
           case KEY_RETURN:
                   bios_scan=0x1c; ascii=0x0d; break;
-          case KEY_F1...KEY_F10:
+          case KEY_F1 ... KEY_F10:
                   bios_scan=key+(0x54-KEY_F1); break;
           case KEY_F11:
                   bios_scan=0x87; break;
--- dosemu-1.0.1/src/tools/periph/dexeconfig.c.orig	Tue Jul 25 20:10:30 2000
+++ dosemu-1.0.1/src/tools/periph/dexeconfig.c	Tue Jul 25 20:10:50 2000
@@ -5,6 +5,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
--- dosemu-1.0.1/src/base/keyboard/keymaps.c.orig	Tue Jul 25 20:37:19 2000
+++ dosemu-1.0.1/src/base/keyboard/keymaps.c	Tue Jul 25 20:37:47 2000
@@ -7,6 +7,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
 #include <linux/kd.h>
 #include <linux/keyboard.h>
 #include <sys/ioctl.h>
